File: src/main/kotlin/com/xs0/gqlktx/types/gql/GBaseType.java
package com.xs0.gqlktx.types.gql;

public abstract class GBaseType extends GType {
    private final String name;

    protected GBaseType(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    @Override
    public String getGqlTypeString() {
        return name;
    }

    @Override
    public GBaseType getBaseType() {
        return this;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        toString(sb);
        return sb.toString();
    }

    public abstract void toString(StringBuilder sb);
}

File: src/main/kotlin/com/xs0/gqlktx/types/gql/GBaseType.kt
package com.xs0.gqlktx.types.gql

abstract class GBaseType protected constructor(val name: String) : GType() {
    override val gqlTypeString = name

    override val baseType: GBaseType
        get() = this

    override fun toString(): String {
        var sb = StringBuilder()
        toString(sb)
        return sb.toString()
    }

    abstract fun toString(sb: StringBuilder)
}
